plugins/inventory/lxd.py

```diff
@@ -0,0 +1,1047 @@
+
+    def _connect_to_socket(self):
+        """connect to lxd socket
+
+        Connect to lxd socket by provided url or defaults
+
+        Args:
+            None
+        Kwargs:
+            None
+        Raises:
+            AnsibleError
+        Returns:
+            None"""
+        error_storage = {}
+        url_list = [self.get_option('url'), self.SNAP_SOCKET_URL, self.SOCKET_URL]
+        urls = (url for url in url_list if self.validate_url(url))
+        for url in urls:
+            try:
+                socket_connection = LXDClient(url, self.client_key, self.client_cert, self.debug)
+                return socket_connection
+            except LXDClientException as err:
+                error_storage[url] = err
+        raise AnsibleError('No connection to the socket: {0}'.format(to_native(error_storage)))
+
+    def _get_networks(self):
+        """Get Networknames
+
+        Returns all network config names
+
+        Args:
+            None
+        Kwargs:
+            None
+        Raises:
+            None
+        Returns:
+            list(names): names of all network_configs"""
+        network_configs = self.socket.do('GET', '/1.0/networks')
+        # e.g. {'type': 'sync',
+        #       'status': 'Success',
+        #       'status_code': 200,
+        #       'operation': '',
+        #       'error_code': 0,
+        #       'error': '',
+        #       'metadata': ['/1.0/networks/lxdbr0']}
+        names = []
+        for index in network_configs['metadata']:
+            # e.g. ['', '1.0', 'networks', 'lxdbr0']
+            names.append(index.split('/')[3])
+        return names
+
+    def _get_containers(self):
+        """Get Containernames
+
+        Returns all containernames
+
+        Args:
+            None
+        Kwargs:
+            None
+        Raises:
+            None
+        Returns:
+            list(names): names of all containers"""
+        # e.g. {'type': 'sync',
+        #       'status': 'Success',
+        #       'status_code': 200,
+        #       'operation': '',
+        #       'error_code': 0,
+        #       'error': '',
+        #       'metadata': ['/1.0/containers/udemy-ansible-ubuntu-2004']}
+        containers = self.socket.do('GET', '/1.0/containers')
+        names = []
+        for index in containers['metadata']:
+            # e.g. ['', '1.0', 'containers', 'udemy-ansible-ubuntu-2004']
+            names.append(index.split('/')[3])
+        return names
+
+    def _get_config(self, branch, name):
+        """Get inventory of container
+
+        Get config of container
+
+        Args:
+            str(branch): Name oft the API-Branch
+            str(name): Name of Container
+        Kwargs:
+            None
+        Source:
+            https://github.com/lxc/lxd/blob/master/doc/rest-api.md
+        Raises:
+            None
+        Returns:
+            dict(config): Config of the container"""
+        config = {}
+        if isinstance(branch, tuple):
```

Comment:
instead of pinning this on a `tuple` type, I'd rather allow for `list` as well.
```suggestion
        if isinstance(branch, (tuple, list)):
```